commands.txt

```diff
@@ -1,4 +1,4 @@
-if [ `uname` = 'Darwin' ]; then export SD_CURL_CMD_WRAPPER="eval" && export SD_ZIP_CMD_WRAPPER=""; else export SD_CURL_CMD_WRAPPER="$(echo 'sd-step exec --pkg-version 7.54.1 core/curl')" && export SD_ZIP_CMD_WRAPPER="$(echo 'sd-step exec core/zip')"; fi
+if [ `uname` = 'Darwin' ]; then export SD_ZIP_CMD_WRAPPER=""; else export SD_ZIP_CMD_WRAPPER="$(echo 'sd-step exec core/zip')"; fi
 cd $SD_ARTIFACTS_DIR
 find . -type f -print > manifest.txt
-if [ "$SD_ZIP_ARTIFACTS" = "true" ]; then ZIPFILE=SD_ARTIFACT.zip && find . -type f -exec $SD_ZIP_CMD_WRAPPER zip $ZIPFILE {} + && f=`echo $ZIPFILE | sed "s/^\.\///"`; encoded=$($SD_CURL_CMD_WRAPPER "curl -s -w \"%{url_effective}\" --data-urlencode \"$f\" -G \"\" | sed -e \"s/^\/?//\"") && $SD_CURL_CMD_WRAPPER "curl --http1.1 --retry 3 -H \"Authorization: Bearer $SD_TOKEN\" -H \"Content-Type: text/plain\" -X PUT $STORE_URL/v1/builds/$SD_BUILD_ID/$ARTIFACTS_DIR_SUFFIX/$encoded -T ./\"$f\""; else find . -type f -exec sh -c 'f=`echo $1 | sed "s/^\.\///"`; encoded=$($SD_CURL_CMD_WRAPPER "curl -s -w \"%{url_effective}\" --data-urlencode \"$f\" -G \"\" | sed -e \"s/^\/?//\""); $SD_CURL_CMD_WRAPPER "curl --http1.1 --retry 3 -H \"Authorization: Bearer $SD_TOKEN\" -H \"Content-Type: text/plain\" -X PUT $STORE_URL/v1/builds/$SD_BUILD_ID/$ARTIFACTS_DIR_SUFFIX/$encoded -T ./\"$f\""' -- {} \;; fi; exit_code=$?; if [ $exit_code -eq 56 ]; then exit 0; else exit $exit_code; fi;
+if [ "$SD_ZIP_ARTIFACTS" = "true" ]; then ZIPFILE=SD_ARTIFACT.zip && find . -type f -exec $SD_ZIP_CMD_WRAPPER zip $ZIPFILE {} + && store-cli set $ZIPFILE --type=artifact; else find . -type f -exec store-cli set $1 --type=artifact -- {} \;; fi;
```

Comment:
does store cli version take care of
1.  retry logic ? 
2. Using http1.1 ? We defaulted to 1.1 when we were observing upload errors wrt default http2.0 .

Reply:
@jithin1987  `store-cli` does have retry logic: https://github.com/screwdriver-cd/store-cli/blob/master/sdstore/sdstore.go#L100-L112

Reply:
I also saw an upload error when HTTP/2 was enabled.
> curl: (92) HTTP/2 stream 1 was not closed cleanly: PROTOCOL_ERROR (err 1)

I confirmed  that there is no error with `store-cli` in our environment so far.